packages/sync/src/conflicts/conflictLink.ts

```diff
@@ -0,0 +1,51 @@
+import { ApolloLink } from "apollo-link";
+import { onError } from "apollo-link-error";
+import { GraphQLError } from "graphql";
+import { DataSyncConfig } from "../config/DataSyncConfig";
+import { ConflictResolutionData, strategies } from "./strategies";
+
+export class ConflictLink {
+    private config: DataSyncConfig;
+
+    constructor(config: DataSyncConfig) {
+        this.config = config;
+    }
+
+    /**
+     * Initialise the conflict link
+     * @returns An apollo link capable of detecting errors
+     */
+    public init(): ApolloLink {
+        const link = onError(({ graphQLErrors, operation, forward }) => {
```

Comment:
Antipattern.  Init is not operating on class and not initializing anything in properties.

Reply:
If we extend the link we can include any other link inside. For example:
https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-ws/src/webSocketLink.ts#L49-L53